src/java/org/apache/cassandra/utils/NativeLibrary.java

```diff
@@ -274,6 +284,45 @@ public void trySkipCache(int fd, long offset, int len, String fileName)
         }
     }
 
+    /**
+     * @param buffer
+     * @param length
+     * @param filename -- source file backing buffer; logged on error
+     * <p>
+     * adviseRandom works even on buffers that are not aligned to page boundaries (which is the
+     * common case for how MmappedRegions is used).
+     */
+    public void adviseRandom(MappedByteBuffer buffer, long length, String filename) {
+        assert buffer != null;
+
+        var rawAddress = Native.getDirectBufferPointer(buffer);
+        // align to the nearest lower page boundary
+        var alignedAddress = new Pointer(Pointer.nativeValue(rawAddress) & -PageAware.PAGE_SIZE);
+        // we want to advise the whole buffer, so if the aligned address is lower than the raw one,
+        // we need to pad the length accordingly.  (we do not need to align `length`, Linux
+        // takes care of rounding it up for us.)
+        length += Pointer.nativeValue(rawAddress) - Pointer.nativeValue(alignedAddress);
+
+        try {
+            int result = wrappedLibrary.callPosixMadvise(alignedAddress, length, MADV_RANDOM);
+            if (result != 0) {
+                NoSpamLogger.log(logger,
+                                 NoSpamLogger.Level.WARN,
+                                 10,
+                                 TimeUnit.MINUTES,
+                                 "Failed madvise on file {}. Error: " + wrappedLibrary.callStrerror(result).getString(0),
+                                 filename);
+            }
+        } catch (UnsatisfiedLinkError e) {
+            logger.warn("madvise not supported on this platform.", e);
```

Comment:
what about logging this only once  ?

Reply:
It looks like the pattern in the existing code is to just ignore the exception, presumably because we log on startup in that scenario.  

            logger.error("Failed to link the C library against JNA. Native methods will be unavailable.", e);